src/DynamoCoreWpf/UI/GuidedTour/Guide.cs

```diff
@@ -394,5 +395,42 @@ internal static void CloseWindowOwned(string windowName, Window mainWindow)
             if (findWindow != null)
                 findWindow.Close();
         }
+
+        /// <summary>
+        /// This method will execute a javascript function located in library.hmtl using reflection.
+        /// due that we cannot include a reference to LibraryViewExtensionMSWebBrowser then we need to use reflection to get the types
+        /// </summary>
+        /// <param name="MainWindow">MainWindow in which the LibraryView is located</param>
+        /// <param name="popupInfo">Popup Information about the Step </param>
+        /// <param name="parametersInvokeScript">Parameters for the WebBrowser.InvokeScript() function</param>
+        internal static void ExecuteJSFunction(UIElement MainWindow, HostControlInfo popupInfo, object[] parametersInvokeScript)
+        {
+            const string webBrowserString = "Browser";
+            const string invokeScriptFunction = "InvokeScript";
+
+            //Try to find the grid that contains the LibraryView
+            var sidebarGrid = (MainWindow as Window).FindName(popupInfo.HostUIElementString) as Grid;
+            if (sidebarGrid == null) return;
+
+            //We need to iterate every child in the grid due that we need to apply reflection to get the Type and find the LibraryView (a reference to LibraryViewExtensionMSWebBrowser cannot be added).
+            foreach (var child in sidebarGrid.Children)
+            {
+                Type type = child.GetType();
+                if (type.Name.Equals(popupInfo.WindowName))
```

Comment:
I feel like we can move the library view look up function to a different utility class

Reply:
@QilongTang The method ExecuteJSFunction was moved from Guide.cs to ResourceUtilities.cs.
see the commit: https://github.com/DynamoDS/Dynamo/pull/12312/commits/0397fe711f7a87f84e45c77e4fc9ddd615ff2c2b